### PR TITLE
New_term : Debugging

### PR DESCRIPTION
**Pull request by Krishnamohan-Yerrabilli**

### CODE REVIEW

**Comment by Krishnamohan-Yerrabilli** [comment]
Is there any changes I want to do? 
Feedback appreciated 🙂

**Comment by Krishnamohan-Yerrabilli** [comment]
My apologies, it won't repeat again, for now do I want to do any changes.

**Comment by CathPag** [comment]
No worries. We'll get back to you soon. Thanks for your patience!! :)

File: content/en/Debugging.md
Comment (jihoon-seo) [inline content/en/Debugging.md:4]: For now, [Netlify build fails](https://app.netlify.com/sites/cncfglossary/deploys/62075abab5c1c00008d7e597#L67).

> failed to unmarshal YAML: yaml: found character that cannot start any token

I guess that this is because there are some tab characters (`\t`) in this file:

```suggestion
---
title: Debugging
status: Completed
category: Concept
```

And, currently most filenames of markdown files in [`content/en/`](https://github.com/cncf/glossary/tree/main/content/en) start with lower case, except `TLS` which is a acronym.

So, please consider to rename this file to `debugging.md` ! 😊

**Review by jihoon-seo — COMMENTED** [review]
@Krishnamohan-Yerrabilli Thank you for your contribution! 😊

PTAL!

File: content/en/Debugging.md
Comment (seokho-son) [reply content/en/Debugging.md:4]: @Krishnamohan-Yerrabilli 

Please update the PR according to @jihoon-seo 's suggestions to resolve build failure issue :)
(renaming file, too)

Comment (Krishnamohan-Yerrabilli) [reply content/en/Debugging.md:4]: My class is going on 
After my college is I will fix it thank you :)

Comment (Krishnamohan-Yerrabilli) [reply content/en/Debugging.md]: As per @jihoon-seo's suggestions changes have been made.

**Review by CathPag — COMMENTED** [review]
This is a great start! The content is good but sometimes a little wordy. See my comments/suggestions. I'll also need this to be reviewed by a technical approver.

File: content/en/Debugging.md
Comment (Krishnamohan-Yerrabilli) [reply content/en/Debugging.md]: sure ok

File: content/en/Debugging.md
Comment (Krishnamohan-Yerrabilli) [reply content/en/Debugging.md]: Correct me if I'm wrong :)


''Many times you create a game or a software or a mathematical computation, as
the code is written by humans, as such error occurs, If that happens software will not actually function
as we thought, to locate the source of the problem, programmers use debugging tools.''

File: content/en/Debugging.md
Comment (Krishnamohan-Yerrabilli) [reply content/en/Debugging.md]: Correct me if I'm wrong.


To explain debugging, first, we want to know what is a bug, it was an 
error or fault in a software program, debugging is the process of 
fixing these bug's in the program, this process starts from after 
the program fails to run and ends with successfully solving 
the problem, we do this to get the desired output.

File: content/en/Debugging.md
Comment (Krishnamohan-Yerrabilli) [reply content/en/Debugging.md]: I will add this in the original file if it is ok, any suggestions would be appreciated.
I think beginners can understand, love to hear other reviewers' point of view :)

File: content/en/debugging.md
Comment (seokho-son) [inline content/en/debugging.md:5]:
```suggestion
category: Concept
---
```

@Krishnamohan-Yerrabilli This missing line incurs Hugo build failure.

**Review by seokho-son — CHANGES_REQUESTED** [review]

File: content/en/Debugging.md
Comment (CathPag) [reply content/en/Debugging.md]: As mentioned, these definitions are kept short and sweet: "To explain debugging, first, we want to know what is a bug" are a lot of words, none of which have helped to define what a bug or debugging is."  To define bug, please start with "A bug is..." (or some other simple sentence structure). Keep it simple :)

This is also not very clear and needs rewriting:
> this process starts from after
> the program fails to run and ends with successfully solving
> the problem, we do this to get the desired output.

I think you mean something like: "Debbuging begins when the program fails and ends with successfully problem resolution and desired output." Although, I don't think this needs to go into a definition. But I'll let others chime in.

**Comment by Krishnamohan-Yerrabilli** [comment]
@CathPag 
Oh no, I accidentally clicked Resolve conversation  :(
thank you for your suggestions I made a note of it and I resolve it as soon as possible.

File: content/en/Debugging.md
Comment (CathPag) [reply content/en/Debugging.md]: Here we need to explain how a debugging tool works. Helping developers "not be overwhelmed" does not address the "how debugging tools help"

> Debugging tools provide developers with the needed context to review code and catch errors. By running the program and connecting a debugger, developers can control the flow of the program and instruct it to stop the program at specific points where ...{they suspect the error may be?]. >> needs more info. what exactly do they see and how are errors flagged?

File: content/en/Debugging.md
Comment (JasonMorgan) [reply content/en/Debugging.md]: Other reviewer here. I'd ask you to incorporate @CathPag's feedback before we progress any further.

**Comment by Krishnamohan-Yerrabilli** [comment]
```Code is written by humans, and, as such, inaccuracy will come. When a programmer develops a game, software, or mathematical computation [?] and an error occurs, that program will likely not function as desired (undefined behavior). To identify the source of the problem the developer has to review the code line-by-line. Without context, i.e. variables, data, location in memory, etc., catching errors is quite hard.```

**Comment by CathPag** [comment]
It's ok if you skip that sentence. Let me know when you incorporated the chances so you can continue the conversation.

**Comment by CathPag** [comment]
Just following up on this @Krishnamohan-Yerrabilli

**Comment by Krishnamohan-Yerrabilli** [comment]
@CathPag I'm sorry to say, I'm not able to follow up on the conversation 
please do what you think is right thank you.

**Comment by CathPag** [comment]
Ok, I'll close this and let someone else work on it

**Comment by Krishnamohan-Yerrabilli** [comment]
I'm extremely sorry @CathPag for wasting your team time, it's completely my fault :(

**Pull request closed without merging**
